app/controllers/user_sets_controller.rb

```diff
@@ -22,20 +21,16 @@ def index
       {name: set.name, id: set.id, items_count: set.count}
     end
 
-    respond_to do |format|
-      format.js do
-        render json: {sets: sets}
-      end
-    end
+    render json: {sets: sets}
   end
 
   def create
     if params[:user_set].nil?
-      render nothing: true, status: 500
+      return render nothing: true, status: 500
```

Comment:
Could be a one line 'return render nothing: true, status: 500 if params[:user_set].nil?'